wandb/sdk/launch/_project_spec.py

```diff
@@ -145,43 +145,74 @@ def get_entry_point(self, entry_point: str) -> "EntryPoint":
             return self._entry_points[entry_point]
         return self.add_entry_point(entry_point)
 
-    def _fetch_project_local(self, api: Api) -> None:
+    def _fetch_project_local(self, internal_api: Api) -> None:
         """Fetch a project into a local directory, returning the path to the local project directory."""
         assert self.source != LaunchSource.LOCAL
-        parsed_uri = self.uri
         if utils._is_wandb_uri(self.uri):
-            run_info = utils.fetch_wandb_project_run_info(self.uri, api)
-            if not run_info["git"]:
-                raise ExecutionError("Run must have git repo associated")
-            utils._fetch_git_repo(
-                self.project_dir, run_info["git"]["remote"], run_info["git"]["commit"]
+            source_entity, source_project, source_run_name = utils.parse_wandb_uri(
```

Comment:
oh i meant storing it on the project spec object at creation — is there reason why we don't just parse it immediately and store those things